docs/source/operators/deploy-kubernetes.md

```diff
@@ -15,248 +17,495 @@ The following sample kernel specifications apply to Kubernetes deployments:
+Choose this option if you want to deploy via a [helm](https://helm.sh/) chart. You can customize your deployment using value files - review Configuraation section below for details.
 
-Ensure the `enterprise-gateway` namespace exists:
+### Create the Enterprise Gateway kubernetes service and deployment
+
+You can execute the helm command from the checked-out release of the Enterprise Gateway git [repository](https://github.com/jupyter-server/enterprise_gateway.git):
 
 ```bash
-kubectl create namespace enterprise-gateway
+helm  upgrade --install  enterprise-gateway \
+  jupyter-server/enterprise_gateway/etc/kubernetes/helm/enterprise-gateway \
+   --kube-context [mycluster-context-name] \
+   --namespace [namespace-name]
+
 ```
 
-From anywhere with helm cluster access, create the service and deployment by running helm from a source release or the git repository:
+Alternatively, helm chart tarball is also accessible as an asset on our [release](https://github.com/jupyter-server/enterprise_gateway/releases) page, replace [VERSION] with specific release version you wantr to use:
 
 ```bash
-helm upgrade --install --namespace enterprise-gateway enterprise-gateway etc/kubernetes/helm/enterprise-gateway
+helm  upgrade --install  enterprise-gateway \
+  https://github.com/jupyter-server/enterprise_gateway/releases/download/[VERSION]/jupyter_enterprise_gateway_helm-[VERSION].tgz \
+   --kube-context [mycluster-context-name] \
+   --namespace [namespace-name]
 ```
 
-the helm chart tarball is also accessible as an asset on our [release](https://github.com/jupyter-server/enterprise_gateway/releases) page:
+## Access to Enterprise Gateway from outside the cluster
+
+Take a look at the Kubernetes [documentation](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster-services/#ways-to-connect) on how you can access the Kubernetes service from outside the cluster.
+
+A Kubernetes Ingress is the most user-friendly way of interacting with the service and that is what we will cover in this section.
+If you do not have a Kubernetes Ingress configured on your cluster the easiest way to get access will be using the NodePort service.
+
+## Kubernetes Ingress Setup
+
+##### Prerequisites
+
+- Ingress controller deployed on your Kubernetes cluster. Review the Kubernetes [documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/) for available options.
+- Wildcard DNS record is configured to point to the IP of the LoadBalancer, which frontends your ingress controller
+- Review specific Ingress controller configuration to enable wildcard path support if you are using Kubernetes version < v1.18
+- With Kubernetes v1.18 Ingress uses `PathType` parameter which is set to `Prefix` in the helm chart by default, so no additional configuration is required
+- Refer to your ingress controller documentation on how to setup TLS with your ingress
+
+###### Update Helm deployment to enable ingress
```

Comment:
Is this meant to be a sub-section of Prerequisites or a sibling?  (Should it have a header level or 3 or 4?)
```suggestion
#### Update Helm deployment to enable ingress
```